sql/moz-fx-data-shared-prod/subscription_platform_derived/stripe_subscriptions_history_v1/query.sql

```diff
@@ -200,44 +200,114 @@ plans AS (
   USING
     (product_id)
 ),
+ca_postal_districts AS (
+  -- https://en.wikipedia.org/wiki/Postal_codes_in_Canada#Components_of_a_postal_code
+  SELECT
+    *
+  FROM
+    UNNEST(
+      ARRAY<STRUCT<district STRING, province_code STRING>>[
+        ("A", "NL"),
+        ("B", "NS"),
+        ("C", "PE"),
+        ("E", "NB"),
+        ("G", "QC"),
+        ("H", "QC"),
+        ("J", "QC"),
+        ("K", "ON"),
+        ("L", "ON"),
+        ("M", "ON"),
+        ("N", "ON"),
+        ("P", "ON"),
+        ("R", "MB"),
+        ("S", "SK"),
+        ("T", "AB"),
+        ("V", "BC"),
+        -- District X covers both NT and NU, so we can't tell which one.
+        ("X", NULL),
+        ("Y", "YT")
+      ]
+    )
+),
 customers AS (
   SELECT
     id AS customer_id,
     COALESCE(
       TO_HEX(SHA256(JSON_VALUE(customers.metadata, "$.userid"))),
       JSON_VALUE(pre_fivetran_customers.metadata, "$.fxa_uid")
     ) AS fxa_uid,
-    COALESCE(customers.address_country, pre_fivetran_customers.address_country) AS address_country,
-    customers.shipping_address_country,
+    COALESCE(
+      NULLIF(customers.address_country, ""),
+      pre_fivetran_customers.address_country
+    ) AS address_country,
+    COALESCE(
+      IF(LENGTH(customers.address_state) = 2, customers.address_state, NULL),
```

Comment:
Most billing address state values for US and Canadian customers are two-letter codes, but there are some with state names instead, including some which are clearly not US/Canadian states.  So although this excludes somewhat legit values like "Colorado" for a few customers, I felt it was worth it to exclude clearly wrong values like "Bayamón" and keep the data consistent for BI.